### DISCUSSION TITLE
Release notes (XplatGenerateReleaseNotes) task taking longer time and failing due to timedout

### DISCUSSION THREAD

**Discussion by nishant-sharma121** [rfennell/AzurePipelines#1502] [Q&A]
After reviewing the pipeline/stage and logs, PFB the findings:
We are generating a release note through pipeline for every build. And pipeline is trying to fetch all the information like PR's, Build information, commit information etc. and it is pushing all those information to wiki.
This stage was taking 15 mins earlier but now (for last 4 weeks) it is inconsistent and sometimes failing after 60 /120 mins (as per the timedoutinmin set at task level)
Checked with MS azure team as well, but didn't get much support as it is third party marketplace extension.

kindly suggest any fix.

**Comment by rfennell** [comment]
The problem is for this extension to provide the rich dataset it does it has to make many API calls. This means throttling and timeouts can be an issue. I have raised the issue of timeouts and retries with the Azure DevOps  team and had no useful response. So we are stuck where we are.

All I can think to do is make the task more 'feature flag based', by this I mean provide a set of boolean parameters to allow you to disable different parts of the data gathering process. So you only make the calls you need. This is already partially in place for some opt-in features. 

However, I have no immediate plans to make such a change,
